### PR TITLE
`at::matmul` and `cudaMalloc` issue --  reproducer

### PR DESCRIPTION
**Pull request by samnordmann**
[DO NOT MERGE]
reproducer for Issue
- #2673

### CODE REVIEW

**Comment by csarofeen** [comment]
Could you please post a profile showing the allocations for each iteration?

**Comment by samnordmann** [comment]
> Could you please post a profile showing the allocations for each iteration? Also, could you show the matmul kernels that are executed, would like to see their names.

Have you checked Issue #2673?
I think everything you ask for is in there, lmk otherwise

File: experiment_matmul_cudaMalloc.cpp [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    mat2.push_back(torch::rand({K, N}, device));
+    if (compute_mode == ComputeMode::Unfused) {
+      unreduced_results.push_back(torch::empty({M, K, N}, device));
+    }
+    if (compute_mode == ComputeMode::MatmulOut ||
+        compute_mode == ComputeMode::Unfused) {
+      results.push_back(torch::empty({M, N}, device));
+    }
+  }
+
```
Comment (naoyam) [inline experiment_matmul_cudaMalloc.cpp:158]: Try `cudaDeviceSynchronize()` here.

Comment (samnordmann) [reply experiment_matmul_cudaMalloc.cpp:158]: done, but nothing changes, see https://github.com/NVIDIA/Fuser/issues/2673#issuecomment-2258125490

**Pull request closed without merging**